### PR TITLE
chore(pkg): explicitly include files for release

### PR DESCRIPTION
**Pull request by rgbkrk**
**What kind of change does this PR introduce?**

Build related change

**Did you add tests for your changes?**

N/A

**If relevant, did you update the documentation?**

N/A

**Summary**

When running `flow` in a project that uses `webpack-cli`, the latest release is causing flow to choke on a non-valid `json` file:

```
> flow

Launching Flow server for /Users/kylek/code/src/github.com/nteract/nteract-ext
Spawned flow server (pid=10402)
Logs will go to /private/tmp/flow/zSUserszSkylekzScodezSsrczSgithub.comzSnteractzSnteract-ext.log
Monitor logs will go to /private/tmp/flow/zSUserszSkylekzScodezSsrczSgithub.comzSnteractzSnteract-ext.monitor_log
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ node_modules/webpack-cli/.nyc_output/8f27269b188e61a59bbf5827de818c30.json:1:1

Unexpected end of input

     1│
```

Since that file shouldn't be part of the release, I added the `.nyc_output` directory to the npm ignore.

**Does this PR introduce a breaking change?**

Nope.

### CODE REVIEW

**Comment by rgbkrk** [comment]
@dhruvdutt would you say this is a `chore` or a `fix`?

**Comment by dhruvdutt** [comment]
chore

**Comment by rgbkrk** [comment]
Rebased with requested commit style.

**Comment by dhruvdutt** [comment]
@rgbkrk Do check what Even has mentioned in his review on [#354](https://github.com/webpack/webpack-cli/pull/354#pullrequestreview-105193566)

**Comment by rgbkrk** [comment]
Yeah I just added on the ones @ev1stensberg explicitly mentioned, looking into what else is misc though I'm sure you'll be able to rattle off which you don't want more easily as a maintainer.

Do you want `docs` in the packaged release?

Looking again, `jsdoc.json` was already in there, so I'm rebasing again.

**Comment by dhruvdutt** [comment]
@rgbkrk `docs` is also included in npmignore file. 
You can also clean up things from npmignore that are already included in .gitignore like `.vscode`

**Comment by rgbkrk** [comment]
Alright, I've organized the npmignore more like the gitignore.

**Comment by rgbkrk** [comment]
Do you publish from a developer laptop or from CI?

**Comment by ooflorent** [comment]
Why not using `files` field from `package.json`? It would simplify the maintenance.

File: .npmignore [hunk 1]
```diff
@@ -1,19 +1,48 @@
...
+*.snap
+
+# CI/CD
 .appveyor.yml
 .codecov.yml
-*.snap
-build
\ No newline at end of file
+
+# Miscellany
```
Comment (ev1stensberg) [inline .npmignore]: Consider revising this

**Comment by rgbkrk** [comment]
Switched this over to using `files` in the package.json. Note that [`README.md` and certain other files are automatically included](https://docs.npmjs.com/files/package.json#files) and do not have to be explicitly stated in `files`.

**Comment by rgbkrk** [comment]
Just to follow up, even though we're using `files` now:

> > Do you publish from a developer laptop or from CI?

> IMO we publish manually. You can see the list of npm publishers here.

Since a `.npmignore` overrides what's in a `.gitignore`, if a developer has files locally that shouldn't be in the package yet are git ignored, they'll accidentally upload them as part of the package. Some files `npm` makes sure to avoid `.DS_Store`, swap files, etc. Since that's a bit of a hairy mess to figure out continuously, the `files` approach is definitely better.

**Comment by ev1stensberg** [comment]
`webpack-cli @2.0.13`

**Comment by rgbkrk** [comment]
Thanks!

**Pull request merged**